src/common/timing/time-to-interactive.js

```diff
@@ -0,0 +1,32 @@
+import { longTask } from '../vitals/long-task'
+
+export class TimeToInteractive {
+  #ltTimer
+  #latestLT
+  #ltUnsub
+  #startTimestamp
+  #resolver
+  #rejector
+  #longTasks = 0
+  constructor ({ startTimestamp, buffered = false }) {
+    this.#startTimestamp = startTimestamp
+    this.#ltUnsub = longTask.subscribe(({ name, value, attrs }) => {
+      this.#latestLT = attrs.ltStart + value
+      this.#longTasks++
+      this.#setLtTimer()
+    }, buffered)
+
+    return new Promise((resolve, reject) => {
```

Comment:
Don't use return in a constructor. I'm honestly surprised eslint didn't complain about this.

Reply:
something feels gross about an empty constructor with the same logic in a class method but fine :)

Reply:
You could use a static factory method and a private constructor.

Reply:
Oh yea, thats a TS thing. lol .... still could use a factory method.

Reply:
Looking through the code, it looks like you only use the `.then` method and don't actually care that it's a promise.....technically you could just have a public then method.